src/main/scala/com/mwz/sonar/scala/scapegoat/ScapegoatSensor.scala

```diff
@@ -50,20 +48,26 @@ final class ScapegoatSensor(scapegoatReportParser: ScapegoatReportParserAPI) ext
 
   /** Saves the Scapegoat information of a module */
   override def execute(context: SensorContext): Unit = {
-    log.info("Initializing the scapegoat sensor.")
-
-    val reportPath = getScapegoatReportPath(context.config)
-    log.info(s"Loading the scapegoat report file: '$reportPath'.")
-    Try(scapegoatReportParser.parse(reportPath)) match {
-      case Success(scapegoatIssuesByFilename) =>
-        log.info("Successfully loaded the scapegoat report file.")
-        processScapegoatWarnings(context, scapegoatIssuesByFilename)
-      case Failure(ex) =>
-        log.error(
-          "Aborting the scapegoat sensor execution, " +
-          s"cause: an error occurred while reading the scapegoat report file: '$reportPath', " +
-          s"the error was: ${ex.getMessage}."
-        )
+    val settings = context.config()
+
+    if (shouldDisableSensor(settings)) {
```

Comment:
I think here you're re-implementing the [`onlyWhenConfiguration` method from the **SensorDescriptor**](http://javadocs.sonarsource.org/6.7.4/apidocs/org/sonar/api/batch/sensor/SensorDescriptor.html#onlyWhenConfiguration-java.util.function.Predicate-).

Reply:
I didn't know about this - thanks for pointing that out, will look into it.

Reply:
It does the job quite well actually - implemented in 2f12199.

Reply:
:+1: